typescript/infra/src/govern/HyperlaneAppGovernor.ts

```diff
@@ -371,17 +371,37 @@ export abstract class HyperlaneAppGovernor<
       chain,
       account.address,
     );
-    const origin = this.interchainAccount.multiProvider.getChainName(
-      accountConfig.origin,
-    );
     console.info(
       chalk.gray(
         `Inferred call for ICA remote owner ${bytes32ToAddress(
           accountConfig.owner,
-        )} on ${origin} to ${chain}`,
+        )} on ${accountConfig.origin} to ${chain}`,
       ),
     );
 
+    // Ensure that we can derive the ICA using the ISM we're aware of.
+    const ismOverride =
+      this.checker.configMap[chain].ownerOverrides?._icaIsmAddress;
+    const recoveredAccount = await this.interchainAccount.getAccount(chain, {
+      owner: accountConfig.owner,
+      origin: accountConfig.origin,
+      ismOverride,
+    });
+    if (!eqAddress(recoveredAccount, account.address)) {
+      console.error(
+        chalk.red(
+          `Failed to recover the target owner ICA: (chain: ${chain}, remote owner: ${accountConfig.owner}, origin: ${accountConfig}).
+          Used ISM override: ${ismOverride}, recovered account: ${recoveredAccount}. Is the ICA's ISM override correct?
+          Defaulting to manual submission.`,
+        ),
+      );
+      return {
+        type: SubmissionType.MANUAL,
+        chain,
+        call,
+      };
+    }
```

Comment:
so once we have deployed the new icas and uncomment out the relevant code for them in the config, i'm not sure I see how we still submit to the old icas to transfer ownership to the new ones

Reply:
Maybe I'm misunderstanding but I thought transferring would involve:
- changing the owner to the new ICA
- we keep `_icaAddress` as the old ICA, so we're capable of sending a tx via the old ICA to transfer to the new owner

And then after that's executed, we:
- update `_icaAddress` to be the new ICA and `_icaIsmAddress` as needed, so we can submit new txs via this new ICA